data_sinks/mqtt.go

```diff
@@ -83,6 +84,27 @@ func MQTT(conf config.MQTTPublisher) chan<- parser.Measurement {
 				if conf.HomeassistantDiscoveryPrefix != "" {
 					publishHomeAssistantDiscoveries(client, conf, measurement)
 				}
+				if conf.PublishRaw {
+					client.Publish(conf.TopicPrefix+"/"+measurement.Mac+"/temperature", 0, false, strconv.FormatFloat(*measurement.Temperature, 'f', -1, 64))
+					client.Publish(conf.TopicPrefix+"/"+measurement.Mac+"/humidity", 0, false, strconv.FormatFloat(*measurement.Humidity, 'f', -1, 64))
```

Comment:
This doesn't handle `nil` values correctly (crahses). For example, the Pro 2-in-1 doesn't have humidity or pressure measurements, and RuuviBridge parses those as `nil` into the `measurement` such that they shouldn't be published. I would recommend using a few helper functions here such that you'd call for example `safePublishFloat(measurement.Humidity, "/humidity")` and this function would then publish that only if the passed measurement is non-nil. You can find an example of this in the prometheus sink: https://github.com/Scrin/RuuviBridge/blob/master/data_sinks/prometheus.go#L173-L177

Reply:
I looked at the example and started writing it up before I saw your example function name and order of operands, but other than that I've re-implemented this as suggested. The code does look a lot more tidy now in addition to being able to handle nil values.